testing/init.go

```diff
@@ -0,0 +1,272 @@
+	rsender2 := priv2.PublicKey()
+	sender2 := utils.Address(rsender2)
+
+	log.Debug(
+		"generated key",
+		zap.String("addr", sender2),
+		zap.String("pk", hex.EncodeToString(priv2[:])),
+	)
+
+	gen := genesis.Default()
+	gen.MinUnitPrice = chain.Dimensions{1, 1, 1, 1}
+	gen.MinBlockGap = 0
+	gen.CustomAllocation = []*genesis.CustomAllocation{
+		{
+			Address: sender,
+			Balance: 10_000_000,
+		},
+	}
+	genesisBytes, err := json.Marshal(gen)
+	gomega.Expect(err).Should(gomega.BeNil())
+
+	app := &appSender{}
+	for i := range instances {
+		ins := createInstance(app, genesisBytes)
+		instances[i] = ins
+		fmt.Println(*instances[i].Cli)
+	}
+
+	// Verify genesis allocations loaded correctly (do here otherwise test may
+	// check during and it will be inaccurate)
+	for _, inst := range instances {
+		cli := inst.Tcli
+		g, err := cli.Genesis(context.Background())
+		gomega.Ω(err).Should(gomega.BeNil())
+
+		csupply := uint64(0)
+		for _, alloc := range g.CustomAllocation {
+			balance, err := cli.Balance(context.Background(), alloc.Address, ids.Empty)
+			gomega.Ω(err).Should(gomega.BeNil())
+			gomega.Ω(balance).Should(gomega.Equal(alloc.Balance))
+			csupply += alloc.Balance
+		}
+		// For now just skipped metadata, focus on running the tests and make general idea
+		exists, _, supply, _, warp, err := cli.Asset(context.Background(), ids.Empty)
+		gomega.Ω(err).Should(gomega.BeNil())
+		gomega.Ω(exists).Should(gomega.BeTrue())
+		//gomega.Ω(string(metadata)).Should(gomega.Equal(tconsts.Symbol))
+		gomega.Ω(supply).Should(gomega.Equal(csupply))
+		//gomega.Ω(owner).Should(gomega.Equal(utils.Address(ed25519.EmptyPublicKey)))
+		gomega.Ω(warp).Should(gomega.BeFalse())
+	}
+
+	app.instances = instances
+	color.Blue("created %d VMs", Vms)
+}
+
+func createInstance(app *appSender, genesisBytes []byte) Instance {
+	nodeID := ids.GenerateTestNodeID()
+	sk, err := bls.NewSecretKey()
+	gomega.Expect(err).Should(gomega.BeNil())
+
+	l, err := logFactory.Make(nodeID.String())
+	gomega.Expect(err).Should(gomega.BeNil())
+
+	dname, err := os.MkdirTemp("", fmt.Sprintf("%s-chainData", nodeID.String()))
+	gomega.Expect(err).Should(gomega.BeNil())
+
+	networkID := uint32(1)
+	subnetID := ids.GenerateTestID()
+	chainID := ids.GenerateTestID()
+
+	snowCtx := &snow.Context{
+		NetworkID:      networkID,
+		SubnetID:       subnetID,
+		ChainID:        chainID,
+		NodeID:         nodeID,
+		Log:            l,
+		ChainDataDir:   dname,
+		Metrics:        metrics.NewOptionalGatherer(),
+		PublicKey:      bls.PublicFromSecretKey(sk),
+		WarpSigner:     warp.NewSigner(sk, networkID, chainID),
+		ValidatorState: &validators.TestState{},
+	}
+
+	toEngine := make(chan common.Message, 1)
+	db := manager.NewMemDB(avago_version.CurrentDatabase)
+
+	v := newController()
+	err = v.Initialize(
+		context.TODO(),
+		snowCtx,
+		db,
+		genesisBytes,
+		nil,
+		[]byte(
+			`{"parallelism":3, "testMode":true, "logLevel":"debug", "trackedPairs":["*"]}`,
```

Comment:
This is pretty hardcoded to `tokenvm`. It should be possible for any VM to provide an arbitrary config.